support custom root types (#628)
samuelcolvin/pydantic: docs/index.rst

.. literalinclude:: examples/schema3.json

> Pydantic models can be set custom root types to support a model that does not have any field exclude ``__root__``.

Reviewer: Can you make this a separate section of the docs. This is not just about schema generation.

Author: I see. I have moved the content to around types sections.

(This script is complete, it should run "as is")

Custom Root Types
.................

> Pydantic models can be set custom root types to support a model that does not have any field exclude ``__root__``.

Reviewer: ```suggestion
Pydantic models which do not represent an "object" in JSON parlance (``dict`` in python) can have a custom
root type defined via the ``__root__`` field. The root type can of any time: list, float, int etc.
```

> You can define custom root types as a type hint of  ``__root__``'s field.

Reviewer: ```suggestion
The root type can be defined via the type hint on the ``__root__`` field.
```

> Also, You can give custom root values to the ``__init__`` method or ``parse_obj`` as a keyword argument.

Reviewer: ```suggestion
The root value can be passed to model ``__init__`` via the ``__root__`` argument as the first and only argument
to ``parse_obj``.
```
